### ISSUE TITLE
Disabled localStorage will immediately throw error

### ISSUE THREAD

**Issue by yumauri** [yumauri/effector-storage#25]
This line will immediately fail if `localStorage` is disabled
https://github.com/yumauri/effector-storage/blob/3587df8c7b7a0dd5595166133dcad160fe37c346/src/local/index.ts#L45

In Chrome:
![image](https://user-images.githubusercontent.com/6583994/127342187-caa02805-ffd5-4833-8504-491288f53275.png)

In Safari on iOS:
![image](https://user-images.githubusercontent.com/6583994/127342281-8ad8ae8b-3fa6-4b55-9259-3773e666cf3d.png)

Same applies to `sessionStorage` as well.

**Comment by yumauri** [comment]
Actually, I'm not sure what correct behaviour should be in that case:

1. Leave this as it is now — runtime error when calling `persist` function.
2. Ignore error and behave like local/session storage doesn't exist at all (like in node environment).
3. Trigger `fail` event once, when calling `persist` function. In that case, if `fail` event will be participant in any logic after calling `persist` function — this logic will not work, because connections with `fail` event are not created yet.
4. Trigger `fail` event every read/write attempt.

Any item, except 1 — is **breaking change**, because some code may already depends on runtime error throwing.

**Comment by yumauri** [comment]
Maybe extend adapters with optional function `supports` (or smth like that)

```javascript
import { persist, supports } from 'effector-storage/local'

const reason = supports()
if (reason != null) {
  console.log('localStorage is not supported:', reason)
}
```

And `persist` will behave like item 2↑ in case adapter is not supported in this particular environment.

**Comment by zerobias** [comment]
Is this will work in that way?

```js
function isLSDefined() {
  try {
    return typeof localStorage !== 'undefined'
  } catch (err) {
    return false
  }
}
```

> Any item, except 1 — is **breaking change**, because some code may already depends on runtime error throwing.

Usually you might imagine code which rely on any throwable case and you'll unable to add even new properties

```js
try {
  persist().anyProperty.toString()
} catch {
  // business logic goes here
  return 'ok'
}
return 'damn effector-storage 😠'
```

**Comment by yumauri** [comment]
Yes, it is possible to catch this error, and go by 2 way — behave like there is no `localStorage` at all.

I thought about smth like this, maybe:

```javascript
function supports() {
  try {
    if (typeof localStorage === 'undefined') {
      return new Error('localStorage is undefined')
    }
  } catch (error) {
    return error
  }
}
```

Function, which could get the reason, without throwing error.
Inside `persist` I can use it and ignore all errors (use `nil` adapter with no-op behaviour) and user could call `supports` manually, if it is needed.

**Comment by yumauri** [comment]
Fixed in [version 4.4.0](https://github.com/yumauri/effector-storage/releases/tag/v4.4.0)
